components/atoms/TextInput/text-input.tsx

```diff
@@ -32,8 +32,10 @@ const TextInput = ({
 
   const handleResetInput = () => {
     handleChange?.("");
-    if (fieldRef) {
+    if (fieldRef?.current) {
       fieldRef.current!.value = "";
+    } else if (inputRef.current) {
+      inputRef.current!.value = "";
```

Comment:
No need for the `!`. The guard `inputRef.current` guarantees that `inputRef.current` exists.

```suggestion
      inputRef.current.value = "";
```